javascript/modules/client/src/client-dao.ts

```diff
@@ -3,34 +3,41 @@ import {
   DaoConfig,
   DaoRole,
   IClientDaoBase,
-  IClientDaoSimpleVote,
-  IClientDaoWhitelist,
-  MintConfig,
-  TokenConfig,
+  IClientDaoERC20Voting, ICreateDaoERC20Voting,
+  IClientDaoWhitelistVoting, ICreateDaoWhitelistVoting,
   VotingConfig,
 } from "./internal/interfaces/dao";
-// import { exampleContractAbi, ExampleContractMethods } from "./internal/abi/dao";
+import {
+  DAOFactory__factory,
+  Registry__factory
+} from "@aragon/core-contracts-ethers";
+
+export { ICreateDaoERC20Voting, ICreateDaoWhitelistVoting }
+
+export class ClientDaoERC20Voting extends ClientCore
+  implements IClientDaoBase, IClientDaoERC20Voting {
 
-export class ClientDaoWhitelist extends ClientCore
-  implements IClientDaoBase, IClientDaoWhitelist {
   /** DAO related methods */
   dao = {
-    create: async (
-      _daoConfig: DaoConfig,
-      _tokenConfig: TokenConfig,
-      _mintConfig: MintConfig,
-      _votingConfig: VotingConfig,
-      _gsnForwarder?: string,
-    ): Promise<string> => {
-      // TODO: This is an ethers.js integration example
-      // const instance = this.attachContractExample(
-      //   "0x1234567890123456789012345678901234567890"
-      // );
-      // const tx = await instance.store("0x1234");
-      // await tx.wait();
+    create: async (params: ICreateDaoERC20Voting): Promise<string> => {
+      if(!this.signer) throw new Error("A signer is needed for creating a DAO");
+      const daoFactoryContract = DAOFactory__factory.connect(this.daoFactoryAddress, this.signer.connect(this.web3));
 
-      // TODO: Not implemented
-      return Promise.resolve("0x1234567890123456789012345678901234567890");
+      const registry = await daoFactoryContract.registry()
+        .then(registryAddress => {
+          return Registry__factory.connect(registryAddress, this.web3)
+        });
+
+      let daoAddress = ''
+      registry.on("NewDAORegistered", (dao: string) => {
+        daoAddress = dao;
+      })
+
+      return daoFactoryContract.newERC20VotingDAO(
+          params.daoConfig, params.votingConfig, params.tokenConfig, params.mintConfig, params.gsnForwarder ?? ''
+      )
+        .then(tx => tx.wait())
+        .then(() => daoAddress);
```

Comment:
Here is where I would make 100% sure that we have a resolved value from the event